### DISCUSSION TITLE
Can code-server support multi-user ?

### DISCUSSION THREAD

**Discussion by jeffliao888** [coder/code-server#6253] [Q&A] [answered]
I want to run code-server in one server for multi-user login accounts, that can be easily to manage the users for me.
I am running  code-server@userA in machineA now,
and running code-server@userB in machineB
running  code-server@userC in machineC

Does any solution for my issue ?  Thank you very much !

**Comment by code-asher** [comment]
There is no multi-user support built into code-server at the moment.  This issue is somewhat related since we would need multi-user support for it: https://github.com/coder/code-server/issues/33.

**Reply by code-asher** [reply]
But you might want to check out https://github.com/coder/coder, you could use it to orchestrate containers for each user each with code-server installed.

**Reply by jeffliao888** [reply, marked as answer]
Dear @code-asher , 
Thank you very much for your help!

I found that if I setup as the following in my machineA server 
userA:  use port 8081 which set in ~/.config/code-server/config.yaml  -> bind-addr: 127.0.0.1:8081
userB:  use port 8082 which set in ~/.config/code-server/config.yaml  -> bind-addr: 127.0.0.1:8082
userC:  use port 8083 which set in ~/.config/code-server/config.yaml  -> bind-addr: 127.0.0.1:8083

after I start the service
>systemctl enable --now code-server@userA
>systemctl enable --now code-server@userB
>systemctl enable --now code-server@userC

then 3 user(userA, userB and userC) can use code-server in one server now, I think I have got my solution already, Thank you !

**Reply by code-asher** [reply]
Awesome!  Yup you can always spin up multiple code-server instances.  

One thing to keep in mind is that if you are binding to a port then users on that same machine can access other user's code-server instances unless you have password authentication on.  

If you are using the built-in password authentication then all is well but if you are relying on external authentication then you might want to bind to a socket owned by each user instead.
